R/S3-methods.R

```diff
@@ -37,13 +37,15 @@ print.sjmisc_frq <- function(x, ...) {
     insight::print_color(sprintf(
       "# total N=%i  valid N=%i  mean=%.2f  sd=%.2f\n\n",
       sum(dat$frq, na.rm = TRUE),
-      sum(dat$frq[1:(nrow(dat) - 1)], na.rm = TRUE),
+      sum(dat$frq[0:(nrow(dat) - 1)], na.rm = TRUE),
       attr(dat, "mean", exact = T),
       attr(dat, "sd", exact = T)
     ), "blue")
 
-    # don't print labels, if all are "none"
-    if (dplyr::n_distinct(dat$label) == 1 && unique(dat$label) == "<none>")
+    # don't print labels, if all except for the NA value are "none"
+    if (dplyr::n_distinct(dat$label[!is.na(dat$val)]) == 1 && unique(dat$label[!is.na(dat$val)]) == "<none>")
+      dat <- dplyr::select(dat, -.data$label)
+    else if (length(dat$val) == 1 && is.na(dat$val))
       dat <- dplyr::select(dat, -.data$label)
 

```

Comment:
Here we have the same action for different if-statements, so they can be ORed together?

Reply:
Yes, as you prefer. I did so for clarity and because the general case will be the first. 
Tell me if you want I do a commit in the PR to change it to a unique `if`.

Reply:
It's a bit more difficult to read, but it doesn't raise confusion. Else, you may think that there _should_ have been a different action, though is actually correct.

Reply:
I have just changed it.